framework/validators/FileValidator.php

```diff
@@ -185,7 +186,12 @@ public function init()
      */
     public function validateAttribute($model, $attribute)
     {
-        if ($this->maxFiles != 1) {
+        $maxFiles = $this->maxFiles instanceof \Closure? call_user_func($this->maxFiles, $model, $attribute): $this->maxFiles ;
```

Comment:
I think `$result = $condition ? $success : $fail;` formatting will be more appropriate in Yii2. Partially it's also related with PSR-2.